docs/source/format/DissociatedIPC.rst

```diff
@@ -0,0 +1,333 @@
+allow for remote memory accessing.
+* Arrow data located on a non-CPU device (such as a GPU) cannot be sent using
+Arrow IPC without having to copy the data back to the host device or copying
+the flatbuffer metadata bytes into device memory.
+  - By the same token, receiving IPC messages into device memory would require
+  performing a copy of the flatbuffer metadata back to the host CPU device.
+
+This protocol is intended to attempt to solve these use cases in an efficient manner.
+
+Goals
+-----
+
+* Define a generic protocol for passing Arrow IPC data, not tied to any particular
+transport, that also allows for utilizing non-CPU device memory, shared memory, and
+newer "high performance" transports such as `ucx`_ or `libfabric`_.
+* Allow for using :ref:`Flight RPC <flight-rpc>` purely for control flow by separating
+the stream of IPC metadata from IPC body bytes
+  - This allows for the data in the body to be kept on non-CPU devices (like GPUs)
+  without expensive Device -> Host copies.
+
+Definitions
+-----------
+
+.. glossary::
+
+   IPC Metadata
+       The flatbuffer message bytes that encompass the header of an Arrow IPC message
+  
+   Tag
+       A ``uint64`` value used as an ID for a message. Specific bits can be masked to
+       allow identifying messages by only a portion of the tag, leaving the rest of the
+       bits to be used for control flow or other message metadata.
+
+   Sequence Number
+       A little-endian, 4-byte unsigned integer starting at 0 for a stream, indicating 
+       the sequence order of messages. It is also used to identify specific messages to 
+       tie the IPC metadata header to its corresponding body since the metadata and body
+       can be sent across separate pipes/streams/transports.
+
+       If a sequence number reaches ``UINT32_MAX``, it should be allowed to roll over as
+       it is unlikely there would be enough unprocessed messages waiting to be processed
+       that would cause an overlap of sequence numbers.
+
+       The sequence number serves two purposes: To identify corresponding metadata and 
+       tagged body data messages and to ensure we do not rely on messages having to arrive
+       in order. A client should use the sequence number to correctly order messages as
+       they arrive for processing.
+
+   Backpressure
+       *Currently* this proposal does not specify any way to manage the backpressure of
+       messages to throttle for memory and bandwidth reasons. For now, this will be 
+       **transport-defined** rather than lock into something sub-optimal. 
+
+       As usage among different transports and libraries grows, common patterns will emerge
+       that will allow for a generic, but efficient, way to handle backpressure across
+       different use cases.
+
+       .. note::
+          The current usage and examples have been tested using UCX and libfabric transports
+          so far, but that's all.
```

Comment:
is this meant to say something more like, it is intended to work with other transports too?

Reply:
The first bullet point in the `Goals` section states that it is intended to be a generic protocol not tied to any particular transport. I figured I didn't need to repeat that here, but i'll add it to the note.